cli/test/e2e/package_repo_release_test.go

```diff
@@ -8,52 +8,68 @@ import (
 	"strings"
 	"testing"
 
+	uitest "github.com/cppforlife/go-cli-ui/ui/test"
 	"github.com/stretchr/testify/require"
 )
 
 const (
-	packagesDir = "packages"
+	packagesDir       = "packages"
+	pkgRepoOutputFile = "package-repository.yml"
+	pkgrName          = "testpackagerepo.corp.dev"
 )
 
 func TestPackageRepositoryReleaseInteractively(t *testing.T) {
-	env := BuildEnv(t)
-	logger := Logger{}
-	kappCtrl := Kctrl{t, env.Namespace, env.KctrlBinaryPath, logger}
-
 	cleanUp := func() {
 		os.RemoveAll(workingDir)
 	}
 	cleanUp()
 	defer cleanUp()
 
-	err := os.Mkdir(workingDir, os.ModePerm)
-	if err != nil {
-		t.Fatalf("Unable to create working directory: %s", err.Error())
-	}
+	env := BuildEnv(t)
+	logger := Logger{}
+	kappCtrl := Kctrl{t, env.Namespace, env.KctrlBinaryPath, logger}
+	kappCli := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+	promptOutput := newPromptOutput(t)
+	setup(t)
 
 	logger.Section("Creating a package repository interactively using pkg repo release", func() {
-		promptOutput := newPromptOutput(t)
 
 		go func() {
 			promptOutput.WaitFor("Enter the package repository name")
-			promptOutput.Write("testpackagerepo.corp.dev")
+			promptOutput.Write(pkgrName)
 			promptOutput.WaitFor("Enter the registry url")
 			promptOutput.Write(env.Image)
 		}()
 
-		err = os.Mkdir(filepath.Join(workingDir, packagesDir), fs.ModePerm)
-		if err != nil {
-			t.Errorf("Unable to create packages directory: %s", err.Error())
-		}
-
-		kappCtrl.RunWithOpts([]string{"pkg", "repo", "release", "--tty=true", "--chdir", workingDir},
+		kappCtrl.RunWithOpts([]string{"pkg", "repo", "release", "--tty=true", "--chdir", workingDir, "--version", "1.0.0"},
 			RunOpts{NoNamespace: true, StdinReader: promptOutput.StringReader(),
 				StdoutWriter: promptOutput.BufferedOutputWriter(), Interactive: true})
 
 		keysToBeIgnored := []string{"creationTimestamp:", "image"}
 		verifyPackageRepoBuild(t, keysToBeIgnored)
 		verifyPackageRepository(t, keysToBeIgnored)
 	})
+
+	logger.Section(fmt.Sprintf("Installing package repository"), func() {
+		pkgrKappAppName := "test-package-repo-app"
+		cleanUpInstalledPkgRepo := func() {
+			kappCli.RunWithOpts([]string{"delete", "-a", pkgrKappAppName},
+				RunOpts{StdinReader: promptOutput.StringReader(), StdoutWriter: promptOutput.BufferedOutputWriter()})
+		}
+		defer cleanUpInstalledPkgRepo()
+
+		kappCli.RunWithOpts([]string{"deploy", "-a", pkgrKappAppName, "-f", filepath.Join(workingDir, pkgRepoOutputFile), "-c"},
+			RunOpts{StdinReader: promptOutput.StringReader(), StdoutWriter: promptOutput.BufferedOutputWriter()})
+		out, _ := kappCtrl.RunWithOpts([]string{"package", "repository", "get", "-r", pkgrName, "--json"}, RunOpts{})
+
+		output := uitest.JSONUIFromBytes(t, []byte(out))
+		for _, m := range output.Tables[0].Rows {
+			val, ok := m["source"]
+			if ok {
+				require.Contains(t, val, fmt.Sprintf("(imgpkg) (1.0.0) %s", env.Image))
+			}
+		}
```

Comment:
We just want to check the `source` column, but I think having something like [this](https://github.com/vmware-tanzu/carvel-kapp-controller/blob/f17a931d4f01920d4c8aedebe9e138816b78cb42/cli/test/e2e/package_repository_test.go#L88-L96) would still be more readable.

Reply:
Even I thought the same, But cant use `require.Exactly` as sha value in the source will be different everytime

Reply:
I see, maybe use regex to remove it? (No strong opinions though, it's just that ranging over `Rows` felt a bit weird)

Reply:
I would just check if there is only one element and then do a `Contains` check on the stringified `Row`

Reply:
^ +1

Reply:
Adopted